source/src/protocols/protein_grid/ProteinGrid.cc

```diff
@@ -0,0 +1,1493 @@
+	std::string matrix_pdb_name = pdb_name_prefix + "_WholeRatio_" + std::to_string(fullness_ratio_);
+
+
+
+	//if using a sub area, tack that onto the name
+	if ( using_sub_area_ ) {
+		matrix_pdb_name = matrix_pdb_name + "_SubRatio_" + std::to_string(sub_fullness_ratio_);
+
+
+	}
+
+	//tack on end
+	matrix_pdb_name = matrix_pdb_name + ".pdb";
+
+	ms_tr.Trace << "Preparing to make visualization pose for " << matrix_pdb_name << std::endl;
+
+	//create a pose to print to a pdb
+	core::pose::Pose matrix_pose = export_protein_matrix_to_pose();
+
+	//dump the pose to a pdb
+	matrix_pose.dump_pdb(matrix_pdb_name);
+
+	//add a comment to the working_pose_ that notes the matrix file that was made
+	core::pose::add_comment(*working_pose_, "Corresponding space fill matrix file:", matrix_pdb_name);
+}
+
+
+// @brief function to be used to convert a base 10 number to base 62 (as a string with characters represented by upper+lower case letters and digits)
+//used in export_protein_matrix_to_pose to assign a unique name to an atom
+//Due to limitations in atom icoor data, an atom name can be no longer than 4 characters, so this provides 62^4 (~14.7M) unique atom names
+std::string ProteinGrid::base_10_to_base_62(core::Size starting_num)
+{
+	//bool to indicate whether to keep processing the number
+	//stop processing once we fully build the string, which occurs when dividing the current number by 62 is <1 (integer would be 0)
+	bool keep_processing = true;
+
+	//string to hold the base 62 representation of the number
+	//characters represented by base_62_cipher_ vector in the .hh file
+	std::string base_62_number = "";
+
+	core::Size curr_num = starting_num;
+
+	while ( keep_processing )
+			{
+		//take the quotient of the current number by 62
+		core::Size quotient = curr_num / 62;
+
+		//derive the modulus of the current number by 62
+		core::Size mod = curr_num % 62;
+
+		//use the mod value to get the corresponding character from the cipher and append to the string
+		//character appends to the front of the string
+		base_62_number.insert(base_62_number.begin(),utility::code_to_6bit(mod + 1));
+
+		//if the quotient is under 62, get the number from the cipher, otherwise we have to repeat the processing operation
+		if ( quotient < 62 ) {
+			// use the quotient to add to the number, unless the quotient is 0 (no need to add a placeholder 0)
+			if ( quotient != 0 ) {
+				base_62_number.insert(base_62_number.begin(),utility::code_to_6bit(quotient + 1));
+			}
+
+			//we have now fully derived the base 62 number and can stop processing
+			keep_processing = false;
+		} else {
+			//set the quotient to the current number and we continue the operation off of it
+			curr_num = quotient;
+		}
+	}
+
+	return base_62_number;
+}
+
+// @brief default constructor
+//will need to use class functions to seed values for input pose and other potential input data
+//should only use parameterized or copy constructor
+ProteinGrid::ProteinGrid()
+{
+
+}
```

Comment:
If you're not using the default constructor for anything, it's best just to omit it. (As you'll get a compiler error if you ever accidentally try to invoke it.)